networks/monza/monza-config/src/lib.rs

```diff
@@ -1,20 +1,36 @@
-#[derive(Debug, Clone, PartialEq, Eq)]
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub struct Config {
+	#[serde(flatten)]
+	#[serde(default = "Config::default_execution_config")]
 	pub execution_config: maptos_execution_util::config::Config,
```

Comment:
Could be renamed without information loss, to get more concise naming in the TOML schema in case we remove the `flatten` flag above.

```suggestion
	pub execution: maptos_execution_util::config::Config,
```

Reply:
There are two patterns which I thought it might be nice to achieve in future commits.

1. **Always wrap `Config` fields in a globally uniquely named struct**: This improves readability of the TOML and is more like protobuf service standard--which this essentially needs to mimic. (We run into the same kind of problems that lead to the decisions made in protobuf standard.)
2. **`Config` and `Parameters`**: each unit knows exactly what it needs to run--perhaps sometimes in different modes. Each `Config` which has optional fields should `.try_into()` into a target `Parameters`. This not only cleans up unit implementations and separate concerns. But, it also means that we can know if the unit will fail slightly earlier in the process. We don't have to wait until it's trying to bind to an address for example.

Reply:
On point 2:

If I understand the idea correctly, `Config` provides the schema for human-readable configuration (via serde) without enforcing constraints like mandatory fields? And then `impl TryFrom<Config> for Parameters` would validate the parameters, provide domain-specific error diagnostics that would be difficult to shoehorn into serde, etc.

I think the separation of concerns is nice, but I'd like to morph `Parameters` into a builder API for protocol unit instances. We need the programmatic way of constructing the objects that's better than monstrous `new` methods with 6+ arguments.